src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,42 @@
 'use strict';
 
 class Animal {
-  // write your code here
-}
+  static alive = [];
+  health = 100;
 
-class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+
+    Animal.alive.push(this);
+  }
+
+  bite(animal) {
+    if (this instanceof Carnivore && animal.hidden === false) {
+      animal.health -= 50;
+
+      if (animal.health <= 0 && Animal.alive.indexOf(animal) >= 0) {
+        Animal.alive.splice(Animal.alive.indexOf(animal), 1);
```

Comment:
Using `indexOf` and `splice` within a loop can lead to performance issues when dealing with a large number of animals. Consider using the `filter` method to remove the dead animal from the `Animal.alive` array.